src/XGBoostSharp/XGBClassifier.cs

```diff
@@ -183,17 +194,38 @@ public float[] Predict(float[][] data)
         return predictions;
     }
 
+    /// <summary>
+    ///   Predict using the gradient boosted model
+    /// </summary>
+    /// <param name="dMatrix">
+    ///   DMatrix to do predictions on
+    /// </param>
+    /// <returns>
+    ///   Predictions
+    /// </returns>
+    public float[] Predict(DMatrix dMatrix)
```

Comment:
@madskonradsen Where possible, we might as well reuse and call the new DMatrix overload methods from the old float based methods. For both `Predict` and `Fit` methods, and for both the regressor and the classifier implementations.